### ISSUE TITLE
Remove got dependency

### ISSUE THREAD

**Issue by jeluard** [ChainSafe/lodestar#6488]
[Got](https://github.com/sindresorhus/got) has a number of dependencies and doesn't provide significant features over `fetch` as used by lodestar.
Replace its [usages](https://github.com/search?q=repo%3AChainSafe%2Flodestar+%5C%22got%5C%22&type=code) (all in `cli`) by plain `fetch`.

Blocked by https://github.com/ChainSafe/lodestar/issues/6505

**Comment by HiroyukiNaito** [comment]
I pick it

**Comment by HiroyukiNaito** [comment]
@ensi321 

Sorry for the busy schedule

I changed 1 file (filename: file.ts) for trial, then execute following command with no error.

```bash
$ yarn build
$ yarn test:unit
```

Is it no problem like this?

https://github.com/ChainSafe/lodestar/compare/unstable...HiroyukiNaito:lodestar:unstable

**Comment by jeluard** [comment]
Looks like the right direction! We probably want to create some small utility function so that no to repeat the same lines everywhere `got` is used?

Also, no need to introduce `Buffer` here. `new Uint8Array(await res.arrayBuffer())` should do the job.

**Comment by HiroyukiNaito** [comment]
@jeluard 
Thanks for feedback!
Does it mean to use it? packages/api/src/utils/client/fetch.ts

**Comment by nflaig** [comment]
Please open a PR to get feedback on code, can be draft until it is ready

**Comment by HiroyukiNaito** [comment]
@nflaig Sure, thanks!. I will.

**Comment by HiroyukiNaito** [comment]
I accidendally unassigned to the issue, but still work on it.

**Comment by HiroyukiNaito** [comment]
`Remove got dependency` cases following another issue when putting command ` ./lodestar dev --network dev`.

```bash
$ ./lodestar dev --network dev
file:///[dir]/lodestar/packages/params/lib/setPreset.js:27
        throw Error(`Lodestar preset is already frozen. You must call setActivePreset() at the top of your
              ^

Error: Lodestar preset is already frozen. You must call setActivePreset() at the top of your
application entry point, before importing @lodestar/params, or any library that may import it.

`
// index.ts
import {setActivePreset, PresetName} from "@lodestar/params/setPreset"
setActivePreset(PresetName.minimal)
// Now you can safely import from other paths and consume params
import {SLOTS_PER_EPOCH} from "@lodestar/params"
console.log({SLOTS_PER_EPOCH})
`

    at setActivePreset (file:///[dir]/lodestar/packages/params/lib/setPreset.js:27:15)
    at file:///[dir]/lodestar/packages/cli/lib/applyPreset.js:41:9
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async file:///[dir]/lodestar/packages/cli/bin/lodestar.js:3:1
``` 

# Problem Cause

packages/cli/src/utils/file.ts will remove got devendency and replace wrapedFetch (@lodestar/api).
@lodestar/api package is imported in packages/cli/src/utils/file.ts which is used when applying the preset.
The problem is that @lodestar/api is imported in packages/cli/src/utils/file.ts which causes the issue because  import that file into packages/cli/src/applyPreset.ts which is not allowed (see warning on top of file)

**Comment by jeluard** [comment]
@HiroyukiNaito Thank you for the follow up. Let me get back to you about this.

**Comment by jeluard** [comment]
Unfortunately it looks like we will have to put your work on pause for now, so that we can find a proper solution to this issue.
@HiroyukiNaito Thanks a lot for your effort on this, it was definitely appreciated.

**Comment by HiroyukiNaito** [comment]
@jeluard Sure. So, I try another one.

**Comment by VictoriaAde** [comment]
Hello, has a solution been found for this issue?
I noticed @nflaig mentioned #6505, is that the solution?

**Comment by matthewkeil** [comment]
Hi @VictoriaAde !!  Yes this is still an active issue.  There is a branch that was attempted by someone that will be a good resource to start with.  That branch has a circular dependency though so we suggest moving the fetch code to the `utils` package

**Comment by matthewkeil** [comment]
> Hello, has a solution been found for this issue? I noticed [@nflaig](https://github.com/nflaig) mentioned [#6505](https://github.com/ChainSafe/lodestar/issues/6505), is that the solution?

Yes, I think #6505 is a decent reference and you can also look to this attempt for some ideas on how to finalize the fix

https://github.com/ChainSafe/lodestar/compare/unstable...HiroyukiNaito:lodestar:unstable

**Comment by VictoriaAde** [comment]
Thank you, @matthewkeil. I will begin working on it—please assign it to me.

**Comment by VictoriaAde** [comment]
> Hi [@VictoriaAde](https://github.com/VictoriaAde) !! Yes this is still an active issue. There is a branch that was attempted by someone that will be a good resource to start with. That branch has a circular dependency though so we suggest moving the fetch code to the `utils` package

Hello, when you said to move fetch to the `utils` package, did you mean packages/cli/src/utils or packages/utils? I couldn't find any use of `got` in packages/utils. 
The previous attempt was in packages/cli/src/utils/file.ts.

**Comment by nflaig** [comment]
> when you said to move fetch to the utils package, did you mean packages/cli/src/utils or packages/utils?

it needs to be part of `@lodestar/utils`, located here `packages/utils` as we need to reuse it in multiple places across the code base, not just the cli

**Comment by VictoriaAde** [comment]
> > when you said to move fetch to the utils package, did you mean packages/cli/src/utils or packages/utils?
> 
> it needs to be part of `@lodestar/utils`, located here `packages/utils` as we need to reuse it in multiple places across the code base, not just the cli

Alright, this makes sense, as I was already thinking in this direction.
